src/libos/src/signal/do_sigtimedwait.rs

```diff
@@ -12,9 +12,9 @@ pub async fn do_sigtimedwait(interest: SigSet, timeout: Option<&Duration>) -> Re
         "do_rt_sigtimedwait: interest: {:?}, timeout: {:?}",
         interest, timeout,
     );
-    // TODO: support timeout
+
     if let Some(timeout) = timeout {
-        warn!("do not support timeout yet");
+        crate::time::do_timeout(timeout);
```

Comment:
I don't think this commit gives set a good example of how timeout can be supported in general. There must be some kind of more high-level abstraction that supports waiting with a timeout, instead of using io_uring's timeout directly.

Reply:
Interface examples:

``` Rust
let timer = Timer::new(io_uring);
let firer = timer.firer();
...
timer.sleep(duration).await
...
timer.sleep_forever().await
...
// firer.fire() ? 
// Should we guarantee that the cancel will be finished in this case?
firer.fire().await
```

``` Rust
if let Err(_) = timeout(duration, future).await {
    println!("the future did not complete within the duration");
}
```

Reply:
1. Should we support Instant interface in timer? e.g., `timer.sleep_at(instant)`.
2. nanosleep(), how to wake by signal?